src/index.js

```diff
@@ -119,8 +120,18 @@ function(err, results) {
     shell.exec('npm version --no-git-tag-version ' + oldVersion);
   }
 
-  // ### STEP 8 - Run if any pre commit script has been specified (DESTRUCTIVE OPERATION)
+  // ### STEP 8 - Run if any pre commit script has been specified (DESTRUCTIVE OPERATION?)
   lib.runPreCommitScript(program.preCommit, version);
+
   // ### STEP 9 - Tag and push (DESTRUCTIVE OPERATION)
-  if (!program.dryrun) lib.addFilesAndCreateTag(version);
+  let pushResultCode = 1; // Default to failure signal
+
+  if (!program.dryrun) {
+    pushResultCode = lib.addFilesAndCreateTag(version);
+  }
+
+  // ### STEP 10 - Run after successful push (DESTRUCTIVE OPERATION?)
+  if (pushResultCode === 0 || program.dryrun) {
```

Comment:
Thanks fro the PR Brett. Do you intent to run `post-success` even for dryrun mode? As we can't control the content of `post-success` or any external script, it may have or cause side-effects (destructive). Dryrun was designed to be a quick way to test the execution without causing file changes or side-effects.